tests/jballerina-unit-test/src/test/resources/test-src/expressions/varref/record-variable-reference.bal

```diff
@@ -369,6 +369,40 @@ function testMappingBindingWithSingleNameFieldBinding() {
     assertEquality("bar4", b);
 }
 
+function testMappingBindingPatternInListBindingPattern() {
+    [record {int a; int b;}] r1 = [{a: 1, b: 2}];
+    [record {int a; int b; stream<int> c;}] r2 = [{a: 1, b: 2, c: new}];
+    [Some, Some] r3 = [{var1: "A", var2: "B"}, {var1: "C", var2: "D"}];
+
+    int a;
+    int b;
+
+    [{a, b}] = r1;
+    [record {int a; int b;}] v = r1;
+
+    string xVar1;
+    string xVar2;
+
+    [{var1: xVar1}, {var1: xVar2}] = r3;
+    [{a, b}] = r2;
+
+    string fname;
+    string lname;
+    string id;
+    int age;
+    string nextId;
+
+    [{id, fname, lname}, {id: nextId, age}] = [{id: "u1001", fname: "John", lname: "doe"}, {id: "xVar", age: 10}];
+
+    int p1;
+    int p2;
+    int p3;
+    int p4;
+    int p5;
+
+    [[{p1, p2}], [[{p3, p4}], {p5}]] = [[{p1: 10, p2: 12}], [[{p3: 13, p4: 14}], {p5: 15}]];
```

Comment:
Add tests for mapping binding patterns in 

i. mapping binding patterns - not directly related to this fix because how it is type checked, but goes through the modified path
ii. error binding patterns - this fails at the moment due to a different issue (https://github.com/ballerina-platform/ballerina-lang/issues/40312), but let's add a simple test because it compiles now and keep it disabled/commented out

```ballerina
function fn(error<record { record { int a; } x; }> r) {
    int a;
    error(x = {a}) = r;
}
```